### DISCUSSION TITLE
Regarding Chi2 definition

### DISCUSSION THREAD

**Discussion by Simple-Shyam** [acts-project/acts#1624] [Q&A] [answered]
Dear ACTS experts, I would like to ask a very technical question. We have five local parameters, Can I ask in which class the Chi2 is defined using these parameters. I would like to understand, how Chi2 is evaluated in these coordinate? 
Can you please point out to the class where we define the Chi2 correspond to the track parameters after smoothing step of Kalman filter?

Thank you in Advance
Shyam

**Comment by paulgessinger** [comment, marked as answer]
Hey @Simple-Shyam, I'm not exactly sure if we calculate the smoothed $\chi^2$ anywhere right now. The `GainMatrixUpdater` contains a calculation of the filtered $\chi^2$, and I believe that is also what is used in the downstream output writers.

Is your question specifically about the smoothed $\chi^2$ or the track(state) $\chi^2$ in general?

**Reply by Simple-Shyam** [reply]
Thank you  paulgessinger,
Yes, I want to understand where is this Chi2 is evaluated corresponding to smoothed hit point? Normally, we minimize these parameters (local 5 parameters) using the measurement on the detector planes by evaluating full covariance matrix of errors. I am using it for (Electron Ion Collider, USA) tracking so want to know so that I can understand some of the details. I want one more thing, for the material map we are using .cbor file. Can we directly supply geometry and it picks from there automatically using geomanager as Genfit does?

**Reply by paulgessinger** [reply]
The Kalman Fitter does not currently calculate the smoothed $\chi^2$. The Combinatorial Kalman Filter uses the local **filtered** $\chi^2$ to select eligible measurements.
This could be added to the KF for diagnostics but is not needed for the fit itself. There is a Global $\chi^2$ fitter in the works (#1580), but I don't believe that uses the local $\chi^2$ either, and "smoothed" doesn't mean anything in that context.

I don't know about the details of Genfit, unfortunately. The material map is produced by a special mapping workflow that uses Geant4 to probe the passive material in your simulation geometry.

Aside: In ACTS we have 6 local parameters, including time.

**Reply by Simple-Shyam** [reply]
Thank you, now it's clear. I asked during a presentation on ACTS in ALICE experiment about this time parameter but it was answered it's not for time based pattern recognition, it's just for truth seeding.

**Reply by paulgessinger** [reply]
It can be, but the current seeding doesn't use it.

**Reply by Simple-Shyam** [reply]
Thanks you

**Comment by asalzburger** [comment]
Dear Shyam,

can you please specify what you would like to compute?

- the chi2 of your 5 parameters w.r.t to a 5 dimensional measurement ?
- the fit chi2/(n.d.f.) of your 5 parameters as estimated by the track fitter ?

**Reply by Simple-Shyam** [reply]
Dear Andreas,
I am involved in the EIC (Electron Ion Collider, USA) track finding and fitting. I have an interpretation that we minimize these five parameters after smoothed hit points. There should be some place, where the chi2 of your 5 parameters w.r.t to a 5 dimensional measurement will be defined. This information can be helpful for me.

**Reply by paulgessinger** [reply]
You should be able to calculate this from the individual track states that you get after the fit.

**Reply by Simple-Shyam** [reply]
Thank yes, I will do. I want to see the formula how it is defined for a helix with 5 local coordinates on each detector planes as in the case of ACTS? Is it some where in the documentation?

**Reply by Simple-Shyam** [reply]
> You should be able to calculate this from the individual track states that you get after the fit.

Thanks I got how to evaluate it.
